dbuild/__init__.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import os
+import shutil
+import glob
+
+from jinja2 import Environment, FileSystemLoader
+from docker import Client
+from tempfile import mkdtemp
+
+def docker_client(url='unix://var/run/docker.sock'):
+    """ return docker client """
+    return Client(url)
+
+def build_image(docker_client, path, tag):
+    """ Build docker image"""
+    return [ ''.join(line.values()).strip() for line in docker_client.build(
+        path=path, rm=True, forcerm=True, tag=tag, decode=True)]
+
+def create_container(docker_client, image, name=None, command=None, env=None,
+        disable_network=False, shared_volumes=None, cwd=None):
+    """ create docker containers """
+    if shared_volumes:
+        volumes = shared_volumes.values()
+        binds = ['{}:{}'.format(k,v) for k,v in shared_volumes.iteritems()]
+        host_config=docker_client.create_host_config(binds=binds)
+    else:
+        host_config=None
+
+    container = docker_client.create_container(image=image, name=name, command=command,
+                            environment=env, network_disabled=disable_network,
+                            volumes=volumes, working_dir=cwd,
+                            host_config=host_config)
+    return container
+
+def start_container(docker_client, container):
+    """ Start docker container """
+    response = docker_client.start(container=container.get('Id'))
+    return response
+
+def wait_container(docker_client, container):
+    """ Wait the container to finish execution """
+    rv = docker_client.wait(container=container)
+    return rv
+
+def container_logs(docker_client, container):
+    """ Get container stdout and stderr """
+    return [log.strip() for log in docker_client.logs(container=container,
+        stream=True, timestamps=True)]
+
+def remove_container(docker_client, container, force=False):
+    """ Remove docker container """
+    return docker_client.remove_container(container=container, force=force)
+
+def docker_build(docker_path, build_dir, build_type, source_dir='source',
+        force_rm=False, docker_url='unix://var/run/docker.sock', flavor='ubuntu', dist='trusty'):
+    c = docker_client(docker_url)
+    print "Starting %s Package Build" % build_type
+    image_tag='dbuild-' + flavor + '/' + dist
+    response = build_image(c, docker_path, tag=image_tag )
+    print '\n'.join(response)
+
+    if build_type == 'source':
+        command = ['dpkg-buildpackage', '-S', '-nc', '-uc', '-us']
+        cwd = '/build/' + source_dir
+    elif build_type == 'binary':
+        command = ['bash', '-c', "dpkg-source -x /build/*.dsc /build/pkgbuild/ && cd /build/pkgbuild && /usr/lib/pbuilder/pbuilder-satisfydepends && dpkg-buildpackage"]
+        cwd = '/build'
+    else:
+        raise Exception('Unknown build_type %s' % build_type)
```

Comment:
You should create a different exception class, inheriting from Exception. Otherwise, there's no way to catch it without also catching every other exception.